android/src/main/java/com/reactnativeone/OneModule.java

```diff
@@ -72,62 +80,102 @@ public void init(String siteKey, String touchpointUri, String apiKey,
-          );
-        }
-
-        @Override
-        public void onError(@NotNull OneSDKError oneSDKError) {
-          notifyProblem(promise, "" + oneSDKError.getSystemCode(), oneSDKError.getErrorMessage());
+    executor.submit(() -> {
+      try {
+        OneResponse response;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+          response = One.sendInteraction(true, sendInteractionRequest).join();
+        } else {
+          response = One.sendInteractionLegacySupport(true, sendInteractionRequest).join();
         }
-
-        @Override
-        public void onFailure(@NotNull OneAPIError oneAPIError) {
-          notifyProblem(promise, "" + oneAPIError.getHttpStatusCode(), oneAPIError.getErrorMessage());
-        }
-      });
-    }
+        One.processResponse(response);
+        WritableNativeMap responseMap = responseObjectToReadableMap(response);
+        notifyResult(promise, responseMap);
+      } catch (ExecutionException error) {
+        notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+        Log.e(NAME, "[Thunderhead] Send Interaction Completion Error: " + error.getCause());
+      } catch (OneSDKError error) {
+        notifyProblem(promise, Integer.toString(error.getSystemCode()), error.getLocalizedMessage());
+        Log.e(NAME, "[Thunderhead] Send Interaction SDK Error: " + error.getErrorMessage());
+      } catch (OneAPIError error) {
+        notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+        Log.e(NAME, "[Thunderhead] Send Interaction Api Error: " + error.getErrorMessage());
+      } catch (Exception error) {
+        notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+        Log.e(NAME, "[Thunderhead] Send Interaction Error: " + error.getLocalizedMessage());
+      }
+    });
   }
 
   @ReactMethod
-  public void sendProperties(String interaction, ReadableMap propertiesMap) {
+  public void sendProperties(String interactionPath, ReadableMap propertiesMap, final Promise promise) {
     HashMap<String, String> properties = getPropertiesFromReadableMap(propertiesMap);
     final OneRequest sendPropertiesRequest = new OneRequest.Builder()
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
+      .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
       .properties(properties)
       .build();
-
-    final OneCall sendPropertiesCall = One.sendProperties(sendPropertiesRequest);
-    sendPropertiesCall.enqueue(null);
+    executor.submit(() -> {
+      try {
+        OneResponse response;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+          response = One.sendProperties(true, sendPropertiesRequest).join();
+        } else {
+          response = One.sendPropertiesLegacySupport(true, sendPropertiesRequest).join();
+        }
+        One.processResponse(response);
```

Comment:
same concern here about calling process before caller has it.

Reply:
Removed